### DISCUSSION TITLE
How to make MUI Sidebar switch to Drawer mode on tablet devices?

### DISCUSSION THREAD

**Discussion by Thanawit007z** [mui/toolpad#4439] [Help and support]
I would like to know how to make the sidebar in Dashboard Layout switch to Drawer mode for tablet-sized devices (e.g., iPad), similar to how it works on mobile devices. Currently, it only switches to Drawer mode when the screen size is small (mobile). How can I achieve this for tablet-sized screens as well?
Ipad
![image](https://github.com/user-attachments/assets/b9ad88db-2acd-4f94-8ea0-d81cadf92685)
mobile
![image](https://github.com/user-attachments/assets/3967e7ec-ba89-405e-82f4-178be7db6dfb)
![image](https://github.com/user-attachments/assets/2d9803de-7d7b-43eb-ac80-26e9f0176d14)

**Comment by bharatkashyap** [comment]
We're in the process of adding a prop to the DashboardLayout which will allow controlling the sidebar's collapsed state: https://github.com/mui/toolpad/issues/4406 

You should be able to detect any screen size and set the collapsed mode with that prop

**Reply by Thanawit007z** [reply]
Is it currently under development? How long will it take?

**Reply by bharatkashyap** [reply]
It's on the [roadmap](https://github.com/orgs/mui/projects/9) for the version after the next one (so v0.11.0) but keep in mind that the roadmap is dynamic and can change sometimes

**Comment by apedroferreira** [comment]
We could maybe make the mobile breakpoint overridable with a new prop in `DashboardLayout`...

In the meantime, if it doesn't cause other issues, maybe you can try changing breakpoint widths in the theme so that the the drawer navigation shows at a larger width. Maybe something like in this example:

```tsx
const exampleTheme = createTheme({
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 1200,
      lg: 1200,
      xl: 1536,
    },
  },
});
```

**Reply by Thanawit007z** [reply]
I have set the breakpoints as you provided, but the sidebar still looks the same.

**Reply by apedroferreira** [reply]
Sorry, I meant that just as an example.
The actual breakpoints should be something like:

```
breakpoints: {
  values: {
    xs: 0,
    sm: 900,
    md: 900,
    lg: 1200,
    xl: 1536,
  },
}
```

, as the mobile drawer only shows under the `sm` breakpoint.
With these values it would show until the screen is 900px wide.
